app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java

```diff
@@ -101,8 +103,10 @@ public NavDrawerBuilder(final Context context) {
                         }
                     }
                 }
-                this.nav_drawer_options.add(context.getString(R.string.stop_sensor));
-                this.nav_drawer_intents.add(new Intent(context, StopSensor.class));
+                if (getBestCollectorHardwareName() != "G7" || Pref.getBooleanDefaultFalse("engineering_mode")) { // If we are using G7, offer the stop sensor option in engineering mode only
```

Comment:
You should use `equals()` for string comparisons as otherwise it is comparing the object reference rather than the content.

Reply:
Thanks for the guidance.
Done